controllers/raycluster_controller_test.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	routev1 "github.com/openshift/api/route/v1"
+)
+
+func stringInList(l []string, s string) bool {
+	for _, i := range l {
+		if i == s {
+			return true
+		}
+	}
+	return false
+}
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyz")
+var r = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func randSeq(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letters[r.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+var _ = Describe("RayCluster controller", func() {
```

Comment:
A personal comment - do you have a specific reason or preference to use Ginkgo?
I personally find it less readable than "usual" tests, but that may be just my own opinion.

Reply:
Ginkgo is the framework included by default by the OperatorSDK framework. I figured Ginkgo would be best practice

Reply:
ok